src/tuneflow_py/models/song.py

```diff
@@ -83,6 +85,89 @@ def remove_track(self, track_id: str):
                 dep_track.remove_output()
         return track
 
+    def get_structures(self):
+        return [StructureMarker(song=self, proto=structure_proto) for structure_proto in self._proto.structures]
+
+    def get_structure_at_index(self, index: int):
+        if index < 0:
+            index = 0
+
+        if index >= len(self._proto.structures):
+            index = len(self._proto.structures) - 1
+
+        if index == -1:
+            return None
+
+        return StructureMarker(song=self, proto=self._proto.structures[index])
+
+    def get_structure_at_tick(self, tick: int):
+        def tick_to_structure_fn(tick):
+            return StructureMarker(song=self, tick=tick, type=StructureType.UNKNOWN)
+
+        def structure_to_tick_fn(structure):
+            return structure.get_tick()
+
+        index = lower_equal(
+            self.get_structures(),
+            tick_to_structure_fn(tick),
+            key=lambda s: structure_to_tick_fn(s),
+        )
+
+        if index < 0:
+            index = 0
+
+        if index >= len(self._proto.structures):
+            index = len(self._proto.structures) - 1
+
+        if index == -1:
+            return None
+
+        return StructureMarker(song=self, proto=self._proto.structures[index])
+
+    def create_structure(self, tick: int, type: StructureType):
+        structure = StructureMarker(song=self, tick=tick, type=type)
+        self._proto.structures.append(structure._proto)
+        structure._proto = self._proto.structures[-1]
+        if len(self._proto.structures) == 1:
+            # If there is only 1 structure, move it to the start.
+            structure.set_tick(0)
+        self._proto.structures.sort(key=lambda x: x.tick)
+
+    def move_structure(self, structure_index: int, move_to_tick: int):
+        structure = self.get_structure_at_index(structure_index)
+        if not structure:
```

Comment:
For example here, if structure_index is set to 9999 and get_structure_at_index still returns the last structure, this if check would not be able catch that.